listenbrainz/db/msid_mbid_mapping.py

```diff
@@ -79,47 +80,49 @@ def fetch_track_metadata_for_items(items: List[ModelT]) -> List[ModelT]:
         Returns:
             The given list of MsidMbidModel objects with updated track_metadata.
     """
-    msid_item_map, mbid_item_map = {}, {}
+    # it is possible to that multiple items have same msid/mbid. for example, pinning the
+    # same recording twice. since the dict is keyed by mbid/msid the corresponding value
+    # should be a iterable of all items having that mbid
+    msid_item_map, mbid_item_map = defaultdict(list), defaultdict(list)
     for item in items:
         if item.recording_mbid:
-            mbid_item_map[item.recording_mbid] = item
+            mbid_item_map[item.recording_mbid].append(item)
         else:
-            msid_item_map[item.recording_msid] = item
+            msid_item_map[item.recording_msid].append(item)
 
     msid_metadatas = load_recordings_from_msids(msid_item_map.keys())
     for metadata in msid_metadatas:
         msid = metadata["ids"]["recording_msid"]
-        item = msid_item_map[msid]
-        item.track_metadata = {
-            "track_name": metadata["payload"]["title"],
-            "artist_name": metadata["payload"]["artist"],
-            "additional_info": {
-                "recording_msid": msid
+        for item in msid_item_map[msid]:
```

Comment:
Nothing is deleted from MsB ever so don't think this should be an issue. Also, if this is missing we need to fix code elsewhere as well because other places in codebase assume presence of title, artist name.